src/lcm/simulation/simulate.py

```diff
@@ -10,21 +13,22 @@
 from lcm.argmax import argmax
 from lcm.dispatchers import spacemap, vmap_1d
 from lcm.interfaces import InternalModel, StateChoiceSpace
+from lcm.typing import InternalUserFunction, ParamsDict
 
 
 def simulate(
-    params,
-    initial_states,
-    continuous_choice_grids,
-    compute_ccv_policy_functions,
+    params: ParamsDict,
+    initial_states: dict[str, Array],
+    continuous_choice_grids: list[dict[str, Array]],
+    compute_ccv_policy_functions: list[Callable[..., tuple[Array, Array]]],
     model: InternalModel,
-    next_state,
-    logger,
-    solve_model=None,
-    vf_arr_list=None,
-    additional_targets=None,
-    seed=12345,
-):
+    next_state: Callable[..., dict[str, Array]],
+    logger: logging.Logger,
+    solve_model: Callable[..., list[Array]] | None = None,
```

Comment:
For the next refactoring: I have a fairly strong opinion that we should have another function `solve_and_simulate`, which
- removes this keyword argument
- gets rid of the `None` option in `pre_computed_vf_arr_list` (then we can change to `value_function_arrays` and make that a tuple).

Also, we should get rid of the default for `seed`.